source/grid/grid_tools_cache.cc

```diff
@@ -110,6 +127,27 @@ namespace GridTools
   }
 #endif
 
+#ifdef DEAL_II_WITH_MPI
+  /**
+   * Return a reference to the cached global_bounding_boxes object.
+   */
+  template<int dim, int spacedim>
+  const std::vector< std::vector< BoundingBox<spacedim> > >
+  &Cache<dim,spacedim>::get_global_bounding_boxes() const
+  {
+    if ( use_bounding_boxes &&
+         (update_flags & update_global_bounding_boxes) )
+      {
+        auto predicate = GridTools::internal::pred_locally_owned<dim,spacedim>;
+        auto local_bbox = GridTools::compute_mesh_predicate_bounding_box
+                          (*tria, predicate, refinement_level, allow_merge, max_boxes);
+        global_bounding_boxes = Utilities::MPI::all_gather(mpi_comm, local_bbox);
```

Comment:
The function documentation should probably say that this is a function that does communication and that needs to be called by all processors.